From c6bbec50b9140cc44165df956230b726109af9b0 Mon Sep 17 00:00:00 2001
From: Rob Crittenden <rcritten@redhat.com>
Date: Tue, 16 Oct 2012 11:11:26 -0400
Subject: [PATCH] Don't configure a reverse zone if not desired in interactive installer.

A reverse zone was always configured in the interactive installer
even if you answered "no" to the reverse zone question. The only way
to not confiugre it was the --no-reverse option.

https://fedorahosted.org/freeipa/ticket/3161
---
 install/tools/ipa-dns-install    |    2 +-
 install/tools/ipa-server-install |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/install/tools/ipa-dns-install b/install/tools/ipa-dns-install
index 84d1bdc2eb5729896ecb62f65feb11122aacf77d..b0c20c533d559477d4906195a787fda95a61902e 100755
--- a/install/tools/ipa-dns-install
+++ b/install/tools/ipa-dns-install
@@ -213,8 +213,8 @@ def main():
     else:
         reverse_zone = bindinstance.find_reverse_zone(ip)
         if reverse_zone is None and not options.no_reverse:
-            reverse_zone = util.get_reverse_zone_default(ip)
             if not options.unattended and bindinstance.create_reverse():
+                reverse_zone = util.get_reverse_zone_default(ip)
                 reverse_zone = bindinstance.read_reverse_zone(reverse_zone, ip)
 
     if reverse_zone is not None:
diff --git a/install/tools/ipa-server-install b/install/tools/ipa-server-install
index 33e1c257fe4b1d19e98c548875c46707e097d2c2..cc25fb85500b6a97e71c822a882cf6f2aca4f0a0 100755
--- a/install/tools/ipa-server-install
+++ b/install/tools/ipa-server-install
@@ -816,8 +816,8 @@ def main():
         if options.reverse_zone:
             reverse_zone = bindinstance.normalize_zone(options.reverse_zone)
         elif not options.no_reverse:
-            reverse_zone = util.get_reverse_zone_default(ip)
             if not options.unattended and bindinstance.create_reverse():
+                reverse_zone = util.get_reverse_zone_default(ip)
                 reverse_zone = bindinstance.read_reverse_zone(reverse_zone, ip)
 
         if reverse_zone is not None:
@@ -839,7 +839,7 @@ def main():
         print "Forwarders:    %s" % ("No forwarders" if not dns_forwarders \
                 else ", ".join([str(ip) for ip in dns_forwarders]))
         print "Reverse zone:  %s" % ("No reverse zone" if options.no_reverse \
-                else reverse_zone)
+                or reverse_zone is None else reverse_zone)
         print
 
     if not options.unattended and not user_input("Continue to configure the system with these values?", False):
-- 
1.6.0.6

